### PR TITLE
ref: upgrade pytest to 7.x

### PR DESCRIPTION
**Pull request by asottile-sentry**
only took about 50 PRs to get warning clean -- but I think we're there now \o/

### CODE REVIEW

File: src/sentry/utils/pytest/relay.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ def relay_server_setup(live_server, tmpdir_factory):
         datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S_%f")
     )
     config_path = tmpdir_factory.mktemp(prefix)
+    config_path.chmod(0o755)
```
Comment (asottile-sentry) [inline src/sentry/utils/pytest/relay.py:55]: pytest temp directories changed to be 0o700 -- this made them unreadable in the mounted container by the unprivileged user

**Review by mattgauntseo-sentry — APPROVED** [review]

**Review by joshuarli — APPROVED** [review]

**Comment by asottile-sentry** [comment]
bleh the getsentry acceptance tests do something wonky -- will have to wait to merge until I fix those :/

**Review by mattgauntseo-sentry — APPROVED** [review]

**Pull request merged**